src/app/components/sandbox/Preview/index.js

```diff
@@ -56,20 +57,27 @@ type State = {
 };
 
 export default class Preview extends React.PureComponent {
+  initialPath: string
+  
   constructor(props: Props) {
     super(props);
 
     this.state = {
       frameInitialized: false,
       history: [],
       historyPosition: 0,
-      urlInAddressBar: '',
+      urlInAddressBar: props.initialPath || '',
       url: null,
     };
 
     if (!props.noDelay) {
       this.executeCode = debounce(this.executeCode, 800);
     }
+
+    // we need a value that doesn't change when receiving `initialPath`
+    // from the query params, or the iframe will continue to be re-rendered
+    // when the user navigates the iframe app, which shows the loading screen
+    this.initialPath = this.state.urlInAddressBar
```

Comment:
Do you mean with this navigating inside the iframe, or manually entering the url?